src/inmanta/data/dataview.py

```diff
@@ -1268,16 +1269,36 @@ def allowed_filters(self) -> dict[str, type[Filter]]:
         """
         Return the specification of the allowed filters, see FilterValidator
         """
-        return {}
+        return {
+            "managed": BooleanEqualityFilter,
+        }
 
     def get_base_url(self) -> str:
         return "/api/v2/discovered"
 
     def get_base_query(self) -> SimpleQueryBuilder:
         query_builder = SimpleQueryBuilder(
-            select_clause="SELECT environment, discovered_resource_id, values",
-            from_clause=f" FROM {data.DiscoveredResource.table_name()}",
-            filter_statements=["environment = $1"],
+            select_clause="SELECT *",
+            prelude=f"""
+            WITH result AS (
+                SELECT
+                    dr.environment,
+                    dr.discovered_resource_id,
+                    dr.values,
+                    (
+                        CASE
+                            WHEN rps.resource_id IS NOT NULL THEN true
+                            ELSE false
+                        END
+                    ) AS managed
+
+                FROM {data.DiscoveredResource.table_name()} as dr
+                LEFT JOIN {data.ResourcePersistentState.table_name()} rps
+                ON dr.discovered_resource_id = rps.resource_id
```

Comment:
I think you need to include the environment as well. Otherwise PostgreSQL might not be able to use the index when performing the join. Did you check the performance of this query?
```suggestion
                LEFT JOIN {data.ResourcePersistentState.table_name()} rps
                ON dr.environment=rps.environment AND dr.discovered_resource_id = rps.resource_id
```

Reply:
It looks better indeed:
before:
```
                                                                       QUERY PLAN
--------------------------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=27.10..27.12 rows=7 width=81) (actual time=0.098..0.104 rows=6 loops=1)
   ->  Sort  (cost=27.10..27.12 rows=7 width=81) (actual time=0.096..0.100 rows=6 loops=1)
         Sort Key: dr.discovered_resource_id
         Sort Method: quicksort  Memory: 25kB
         ->  Hash Right Join  (cost=12.69..27.01 rows=7 width=81) (actual time=0.058..0.070 rows=6 loops=1)
               Hash Cond: ((rps.resource_id)::text = (dr.discovered_resource_id)::text)
               ->  Seq Scan on resource_persistent_state rps  (cost=0.00..13.40 rows=340 width=32) (actual time=0.005..0.007 rows=4 loops=1)
               ->  Hash  (cost=12.64..12.64 rows=4 width=80) (actual time=0.038..0.039 rows=6 loops=1)
                     Buckets: 1024  Batches: 1  Memory Usage: 9kB
                     ->  Bitmap Heap Scan on discoveredresource dr  (cost=4.18..12.64 rows=4 width=80) (actual time=0.028..0.031 rows=6 loops=1)
                           Recheck Cond: (environment = '1a2d7489-c7dc-4e94-b0c1-1a35fd5170ba'::uuid)
                           Heap Blocks: exact=1
                           ->  Bitmap Index Scan on discoveredresource_pkey  (cost=0.00..4.18 rows=4 width=0) (actual time=0.016..0.016 rows=6 loops=1)
                                 Index Cond: (environment = '1a2d7489-c7dc-4e94-b0c1-1a35fd5170ba'::uuid)
 Planning Time: 0.372 ms
 Execution Time: 0.160 ms

```

After:
```
                                                                                      QUERY PLAN
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=22.20..22.24 rows=4 width=81) (actual time=0.081..0.096 rows=6 loops=1)
   ->  Merge Left Join  (cost=22.20..22.24 rows=4 width=81) (actual time=0.079..0.092 rows=6 loops=1)
         Merge Cond: ((dr.discovered_resource_id)::text = (rps.resource_id)::text)
         Join Filter: (dr.environment = rps.environment)
         ->  Sort  (cost=12.68..12.69 rows=4 width=80) (actual time=0.051..0.052 rows=6 loops=1)
               Sort Key: dr.discovered_resource_id
               Sort Method: quicksort  Memory: 25kB
               ->  Bitmap Heap Scan on discoveredresource dr  (cost=4.18..12.64 rows=4 width=80) (actual time=0.027..0.030 rows=6 loops=1)
                     Recheck Cond: (environment = '1a2d7489-c7dc-4e94-b0c1-1a35fd5170ba'::uuid)
                     Heap Blocks: exact=1
                     ->  Bitmap Index Scan on discoveredresource_pkey  (cost=0.00..4.18 rows=4 width=0) (actual time=0.016..0.016 rows=6 loops=1)
                           Index Cond: (environment = '1a2d7489-c7dc-4e94-b0c1-1a35fd5170ba'::uuid)
         ->  Sort  (cost=9.51..9.52 rows=2 width=48) (actual time=0.020..0.021 rows=4 loops=1)
               Sort Key: rps.resource_id
               Sort Method: quicksort  Memory: 25kB
               ->  Bitmap Heap Scan on resource_persistent_state rps  (cost=4.16..9.50 rows=2 width=48) (actual time=0.008..0.010 rows=4 loops=1)
                     Recheck Cond: (environment = '1a2d7489-c7dc-4e94-b0c1-1a35fd5170ba'::uuid)
                     Heap Blocks: exact=1
                     ->  Bitmap Index Scan on resource_persistent_state_environment_agent_resource_id_idx  (cost=0.00..4.16 rows=2 width=0) (actual time=0.004..0.004 rows=4 loops=1)
                           Index Cond: (environment = '1a2d7489-c7dc-4e94-b0c1-1a35fd5170ba'::uuid)
 Planning Time: 0.369 ms
 Execution Time: 0.165 ms
```
No seq scan when joining on env as well